### ISSUE TITLE
Make favicon service configurable, or with a relative url

### ISSUE THREAD

**Issue by EricGrange** [keeweb/keeweb#1875]
When installing keeweb to your own website (using the html distribution), the webapp still uses hard-coded (AFAICT) https://services.keeweb.info/favicon/ to fetch the icon. But that URL is blocked by CORS.

It would be nice if the services was configurable (through config.json) or was relative by default (favicon subfolder of the location where keeweb is installed, as it can be easily remapped server-side). The [favicon-proxy](https://github.com/keeweb/favicon-proxy) list the yandex service url (which allows cors), and other alternative free services could be configured/provided later.

Making services.keeweb.info open to CORS would be simpler, but that opens bandwidth and abuse issues (and if it's using the heroku favicon-proxy, that means a potentially very high cost I guess)

**Comment by antelle** [comment]
Hi! This indeed makes sense, also because of privacy reasons (people who're concerned about favicon requests being shared from self-hosted installations), so a setting should be added.

But the service currently allows CORS:

```
curl -s -I https://services.keeweb.info/favicon/apple.com -H Referrer: https://example.com | grep origin
access-control-allow-origin: *
```

**Comment by antelle** [comment]
For example if you try on example.com:

<img width="541" alt="cors" src="https://user-images.githubusercontent.com/633557/129085848-868126e2-2f1a-4297-b74e-4ba17ff0cbb6.png">

**Comment by EricGrange** [comment]
Hmm, I'm seeing CORS errors in Chrome, I had an older release and updated to 1.18.7 and they're still there, for instance

`Access to fetch at 'https://services.keeweb.info/favicon/forum.affinity.serif.com' from origin 'https://REDACTED' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by antelle** [comment]
What is the response code that you got, was it 200?

**Comment by EricGrange** [comment]
No response code, it's blocked by CORS here is the network tab in Chrome

![image](https://user-images.githubusercontent.com/2197210/129086358-f0c61df0-58a6-4803-9503-5fde1df7f252.png)

**Comment by EricGrange** [comment]
For reference yandex favicon fetches the correct icon https://favicon.yandex.net/favicon/forum.affinity.serif.com

**Comment by antelle** [comment]
What do you see inside if you click this failed request?

**Comment by EricGrange** [comment]
![image](https://user-images.githubusercontent.com/2197210/129086801-1b6e0d02-c36f-485e-ab5e-9cc5756f68f3.png)

**Comment by EricGrange** [comment]
![image](https://user-images.githubusercontent.com/2197210/129086835-898d4ce7-b8f0-4670-9da8-45441aba5799.png)

**Comment by antelle** [comment]
Oh, I know why! The favicon can't be loaded, try it in your browser: https://services.keeweb.info/favicon/forum.affinity.serif.com

Then the service doesn't add the CORS headers. Otherwise, if it was successful, they're added. 

I'll fix that, thanks!

**Comment by EricGrange** [comment]
Ok, thanks!

Though providing the ability to use an alternative service would still be very desirable, yandex can fetch the correct favicon https://favicon.yandex.net/favicon/forum.affinity.serif.com for instance (and they support CORS)

**Comment by antelle** [comment]
Yes, I agree, having an option to choose would be very useful, it will be added

**Comment by EricGrange** [comment]
By the way, thanks for your work, it is much appreciated!
